new-components/reporters/elasticsearch/internal/reporter/reporter.go

```diff
@@ -0,0 +1,155 @@
+package reporter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"log/slog"
+
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/esapi"
+	"github.com/go-errors/errors"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/smithy-security/pkg/env"
+
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+
+	"github.com/smithy-security/smithy/sdk/component"
+)
+
+type Conf struct {
+	ElasticsearchURL    string
+	ElasticsearchIndex  string
+	ElasticsearchApiKey string
+}
+
+type esLogger struct {
+	Conf *Conf
+}
+
+// NewJsonLogger returns a new json logger.
+func NewElasticsearchLogger(c *Conf) esLogger {
+	return esLogger{
+		Conf: c,
+	}
+}
+
+// NewConf returns a new configuration build from environment lookup.
+func NewConf(envLoader env.Loader) (*Conf, error) {
+	var envOpts = make([]env.ParseOption, 0)
+	if envLoader != nil {
+		envOpts = append(envOpts, env.WithLoader(envLoader))
+	}
+
+	esURL, err := env.GetOrDefault(
+		"ELASTICSEARCH_URL",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
```

Comment:
⛏️ we return an error by default so you can avoid using the functional option  here